ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -2413,10 +2388,14 @@ protected void onePhaseCommit (boolean reportHeuristics)
                      */
                
                 if (p == TwoPhaseOutcome.ONE_PHASE_ERROR) {
-                   if (recordBeingHandled instanceof ExceptionDeferrer)
-                      onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled;
-                   else if (recordBeingHandled.value() instanceof ExceptionDeferrer) 
-                      onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled.value();
+                   if (recordBeingHandled instanceof ExceptionDeferrer) {
+                       ExceptionDeferrer onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled;
+                       onePhaseCommitExceptionDeferrer.getDeferredThrowables(deferredThrowables);
+                   }
+                   else if (recordBeingHandled.value() instanceof ExceptionDeferrer) {
+                       ExceptionDeferrer onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled.value();
+                       onePhaseCommitExceptionDeferrer.getDeferredThrowables(deferredThrowables);
+                   }
```

Comment:
Code duplication - can you replace these 10 lines with just:

    addDeferredThrowables(recordBeingHandled, deferredThrowables);